fmriprep/cli/sample_openfmri.py

```diff
@@ -0,0 +1,118 @@
+
+    # list subjects
+    cd openfmri
+
+
+"""
+
+import os
+import glob
+import numpy as np
+
+
+def get_parser():
+    """Build parser object"""
+    from argparse import ArgumentParser
+    from argparse import RawTextHelpFormatter
+
+    parser = ArgumentParser(
+        description='OpenfMRI participants sampler, for FMRIPREP\'s testing purposes',
+        formatter_class=RawTextHelpFormatter)
+
+    parser.add_argument('openfmri_dir', action='store',
+                        help='the root folder of a the openfmri dataset')
+
+    # optional arguments
+    parser.add_argument('-D', '--datalad_fetch', action='store_true', default=False,
+                        help='download sampled subjects')
+    parser.add_argument('-o', '--output-file', action='store', help='write output file')
+    parser.add_argument('-n', '--num-participants', action='store', type=int, default=4,
+                        help='number of participants randomly selected per dataset')
+    parser.add_argument('--njobs', action='store', type=int, help='parallel downloads')
+    parser.add_argument('--seed', action='store', type=int, default=20170914,
+                        help='seed for random number generation')
+
+    return parser
+
+
+def main():
+    """Entry point"""
+    thispath = os.getcwd()
+    opts = get_parser().parse_args()
+    np.random.seed(opts.seed)
+
+    out_file = None
+    if opts.output_file is not None:
+        out_file = os.path.abspath(opts.output_file)
+
+    os.chdir(opts.openfmri_dir)
+    all_sub = sorted(glob.glob('ds*/sub-*'))
+    datasets = {}
+    multises = []
+    for subj in all_sub:
+        ds = subj.split('/')[0]
+        if os.path.isdir(os.path.join(subj, 'anat')) and os.path.isdir(os.path.join(subj, 'func')):
+            if ds not in datasets:
+                datasets[ds] = []
+            datasets[ds] += [os.path.basename(subj)]
+            continue
+
+        if (glob.glob(os.path.join(subj, 'ses-*', 'anat')) and
+                glob.glob(os.path.join(subj, 'ses-*', 'func'))):
+            multises.append(ds)
+            if ds not in datasets:
+                datasets[ds] = []
+            datasets[ds] += [os.path.basename(subj)]
+            continue
+
+    multises = list(set(multises))
+    singleses = list(set(datasets.keys()) - set(multises))
+    subsample = {}
+
+    n_sample = 0
+    for ds, sublist in datasets.items():
+        n_sample += min(opts.num_participants, len(sublist))
+        if len(sublist) <= opts.num_participants:
+            subsample[ds] = sublist
+        else:
+            subsample[ds] = sorted(np.random.choice(
+                sublist, size=opts.num_participants, replace=False).tolist())
+
+    if out_file is not None:
+        import yaml
+        with open(out_file, 'w') as outfh:
+            outfh.write(yaml.dump(subsample))
+        print('Sampled participants stored to %s' % out_file)
+
+    all_sub = [sub for _, sublist in datasets.items() for sub in sublist]
```

Comment:
Reused variable name, but only used for `len(all_sub)`, which I think should be the same as `n_sample`?